src/components/users/profile/EditProfile.js

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable react/forbid-prop-types */
+/* eslint-disable react/require-default-props */
+/* eslint-disable react/prefer-stateless-function */
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import ReactFileReader from 'react-file-reader';
+import Navbar from '../../layouts/Navbar';
+import Footer from '../../layouts/Footer';
+import Alert from '../../layouts/Alert';
+import Input from '../../common/Input';
+import Button from '../../common/Button';
+import { getCurrentProfile, updateProfile } from '../../../actions/profile';
+import avatar from '../../../assets/Images/avatar.svg';
+
+export class EditProfile extends Component {
+  state = {
+    username: '',
+    bio: '',
+    image: '',
+    fileImg: ''
+  };
+
+  componentDidMount() {
+    const { getCurrentProfile } = this.props;
+    getCurrentProfile();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.profile) {
+      const { profile } = nextProps;
+
+      profile.username = profile.username !== null ? profile.username : '';
+      profile.bio = profile.bio !== null ? profile.bio : '';
+      profile.image = profile.image !== null ? profile.image : '';
+
+      this.setState({
+        username: profile.username,
+        bio: profile.bio,
+        image: profile.image
+      });
+    }
```

Comment:
what are you trying t achieve here?

Reply:
I was updating the state based on the profile get from the database